olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -70,31 +71,30 @@ olp::client::CancellationToken VersionedLayerClientImpl::GetData(
 }
 
 client::CancellationToken VersionedLayerClientImpl::AddGetDataTask(
-    DataRequest data_request, Callback callback) const {
+    DataRequest request, Callback callback) const {
   auto catalog = catalog_;
   auto layer_id = layer_id_;
   auto settings = settings_;
   auto pending_requests = pending_requests_;
-  auto request_key = pending_requests->GenerateRequestPlaceholder();
-  olp::client::CancellationContext context;
-  olp::client::CancellationToken token(
-      [context]() mutable { context.CancelOperation(); });
 
-  pending_requests->Insert(token, request_key);
+  auto get_data_task = [catalog, layer_id, request,
+                        settings](client::CancellationContext context) {
+    return repository::DataRepository::GetVersionedData(
+        catalog, layer_id, settings, request, context);
+  };
+
+  auto task_context = std::make_shared<TaskContext<DataResult>>(
+      std::move(get_data_task), std::move(callback));
```

Comment:
If we cannot use without a shared_ptr wrapper, please think about implementing it as a shared pimpl.

Reply:
Done